api/commands/build.ts

```diff
@@ -4,18 +4,22 @@ import * as dbadapters from "@dataform/api/dbadapters";
 import { adapters } from "@dataform/core";
 import * as utils from "@dataform/core/utils";
 import { dataform } from "@dataform/protos";
+import { prune } from "df/api/commands/prune";
 
-export function build(
+export async function build(
   compiledGraph: dataform.ICompiledGraph,
   runConfig: dataform.IRunConfig,
-  credentials: Credentials
+  credentials: Credentials,
+  overrideState?: dataform.IWarehouseState
```

Comment:
what's the thinking here?

Reply:
Just for testing, I could change the tests to just test the prune() method directly but that leaves this code untested except for e2e or example tests. Maybe that's fine if you'd prefer it?

Reply:
removing it seems best IMO since this is a public API and we'd rather people don't add dependencies on this parameter.

Reply:
Done!